lib/plottr_import_export/src/importer/scrivener/importer.js

```diff
@@ -645,17 +757,108 @@ function mapMatchedFiles(newJson, matchFiles, fileContents, bookId, isSection) {
         if (newRTFContent.contents && !isSection) {
-          fileContents.rtfContents.contents = newRTFContent.contents
+          fileContents.rtfContents = newRTFContent
         }
       } else {
-        const newRTFContent = getRTFContents(newJson.lines, file, matchedFileId, bookId, isSection)
+        const newRTFContent = getRTFContents(
+          newJson,
+          newJson.lines,
+          file,
+          matchedFileId,
+          bookId,
+          isSection
+        )
         Object.assign(fileContents.rtfContents, newRTFContent)
       }
     }
   })
   return fileContents
 }
 
+function getManuscriptSectionItems(currentJson, RTFContents, bookId) {
+  return RTFContents.map((item, idx) => {
+    if (item && item.children) {
+      const isBold = item.isBold
+      const RTFText = item.children[0].text
+      const isInlineTitle =
+        RTFText.indexOf(':') && sectionItems.includes(RTFText.split(':')[0].toLowerCase().trim())
+      const nextTextValue =
+        RTFContents[idx + 1] &&
+        RTFContents[idx + 1].children[0] &&
+        RTFContents[idx + 1].children[0].text
+          ? RTFContents[idx + 1].children[0].text
+          : '...'
+
+      if (isInlineTitle || isBold) {
+        const strippedInlineTitleText =
+          isInlineTitle && !isBold ? RTFText.split(':')[0].toLowerCase().trim() : ''
+
+        const sectionName = isBold
+          ? RTFText.toLowerCase().trim()
+          : isInlineTitle
+          ? strippedInlineTitleText
+          : null
+
+        // if current item isBold, get next item as section text
+        // if isInlineTitle, get text after colon
+        const sectionText = isBold ? nextTextValue : isInlineTitle ? RTFText.split(':').pop() : null
+        const strippedSectionName = sectionName.toLowerCase().trim()
+        const splittedSectionText = sectionText.includes(',')
+          ? sectionText.split(',')
+          : [sectionText]
+        const sectionIds = splittedSectionText.map((val) => {
+          switch (strippedSectionName) {
+            case i18n('character'):
+            case i18n('characters'):
+              return createNewCharacter(currentJson.characters, { name: val, bookId })
+            case i18n('note'):
+            case i18n('notes'):
+              return createNewNote(currentJson.notes, { title: val, bookId })
+            case i18n('place'):
+            case i18n('places'):
+              return createNewPlace(currentJson.places, { name: val, bookId })
+            case i18n('tag'):
+            case i18n('tags'):
+              return createNewTag(currentJson.tags, val)
+            default:
+              return
+          }
+        })
+        return {
+          [strippedSectionName]: sectionItems.includes(strippedSectionName)
+            ? sectionIds
+            : sectionText,
+        }
+      } else return null
+    } else return null
+  })
+}
+
+function filterRTFContents(RTFContents) {
```

Comment:
This function needs tests and a better name.
The name should describe the intention, not the function we called on `RTFContents`.
How about `keepNonDuplicatedNonEmptyText` ?
Also: what if a user repeats a line verbatim?